src/globus_cli/termio/server_timing.py

```diff
@@ -0,0 +1,154 @@
+    desc_width = (max(len(x[0]) for x in items) if items else 0) + 1
+
+    hborder = click.style(f"+{'-' * (use_width + 2)}+", fg=_BORDER_COLOR)
+    vborder = click.style("|", fg=_BORDER_COLOR)
+    click.echo(hborder, err=True)
+
+    for desc, size in items:
+        desc = desc.ljust(desc_width, ".")
+        bar_width = max(int((use_width - desc_width) * size / factor), 1)
+        bar = "#" * bar_width
+        msg = desc + click.style(bar, fg=_FILL_COLOR)
+        style_char_length = len(msg) - len(click.unstyle(msg))
+        msg = msg.ljust(use_width + style_char_length, " ")
+        click.echo(f"{vborder} {msg} {vborder}", err=True)
+    click.echo(hborder, err=True)
+
+
+class ServerTimingParser(abc.ABC):
+    # which version of the Server-Timing spec does this parser implement?
+    spec_reference: t.ClassVar[str]
+
+    @abc.abstractmethod
+    def parse_single_metric(self, metric_str: str) -> Metric:
+        ...
+
+    def parse_metric_header(
+        self, header_str: str, skip_errors: bool = True
+    ) -> list[Metric]:
+        metric_items = header_str.split(",")
+        ret: list[Metric] = []
+        for item in metric_items:
+            try:
+                ret.append(self.parse_single_metric(item))
+            except ServerTimingParseError:
+                if skip_errors:
+                    pass
+                else:
+                    raise
+        return ret
+
+
+class Draft2017Parser(ServerTimingParser):
+    """
+    Parsing per the Server-Timing draft from 2017 and earlier
+    The spec has changed since this draft.
+
+    For example
+
+      'a=1; "alpha", b=2, c, d; "delta"'
+
+    will parse as
+
+    Metrics:
+      - name: a
+        description: alpha
+        duration: 1.0
+      - name: b
+        duration: 2.0
+      - name: c
+      - name: d
+        description: delta
+    """
+
+    spec_reference = "https://www.w3.org/TR/2017/WD-server-timing-20171018/"
+
+    def parse_single_metric(self, metric_str: str) -> Metric:
+        parts = [p.strip() for p in metric_str.split(";")]
+        if len(parts) == 1:
+            return _parse_simple_metric_part(parts[0])
+        elif len(parts) == 2:
+            metric = _parse_simple_metric_part(parts[0])
+            metric.description = parts[1].strip('"')
+            return metric
+        else:
+            raise ServerTimingParseError(
+                "Too many semicolons in timing item, cannot parse"
+            )
+
+
+def _parse_simple_metric_part(metric: str) -> Metric:
+    metric = metric.strip()
+    if not metric:
+        raise ServerTimingParseError("encountered empty metric")
+
+    if "=" in metric:
+        name, _, unparsed_value = metric.partition("=")
+        try:
+            value = float(unparsed_value)
+        except ValueError as e:
+            raise ServerTimingParseError("Metric value did not parse as float") from e
+        return Metric(name=name, duration=value)
+    else:
+        return Metric(name=metric)
```

Comment:
I prefer the simple case to appear first, and then dedent the rest of the block:

```suggestion
    if "=" not in metric:
        return Metric(name=metric)

    name, _, unparsed_value = metric.partition("=")
    try:
        value = float(unparsed_value)
    except ValueError as e:
        raise ServerTimingParseError("Metric value did not parse as float") from e
    return Metric(name=name, duration=value)
```